lms/views/admin/organization.py

```diff
@@ -0,0 +1,42 @@
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.view import view_config, view_defaults
+
+from lms.models import Organization
+from lms.security import Permissions
+from lms.services import OrganizationService
+
+
+@view_defaults(request_method="GET", permission=Permissions.ADMIN)
+class AdminOrganizationViews:
+    def __init__(self, request):
+        self.request = request
+        self.organization_service: OrganizationService = request.find_service(
+            OrganizationService
+        )
+
+    @view_config(
+        route_name="admin.organization",
+        request_method="GET",
+        renderer="lms:templates/admin/organization.html.jinja2",
+    )
+    def show_organization(self):
+        org = self._get_org_or_404(self.request.matchdict["public_id"])
+        return {"org": org}
+
+    @view_config(
+        route_name="admin.organization",
+        request_method="POST",
+        renderer="lms:templates/admin/organization.html.jinja2",
+    )
+    def update_organization(self):
+        org = self._get_org_or_404(self.request.matchdict["public_id"])
+
+        org.name = self.request.params.get("name", "").strip() or None
```

Comment:
I think I can just about cope with the additional complexity this has introduced in this PR. :)